### PR TITLE
Fixes #340 physical item not being bind-updated 

### LINKED ISSUE #340: Physical items miss binding updates.
### Description

We are observing several defects in iron-data-table that we eventually tracked down to what we think is a defect in iron-list. The problem happens, in our view , because in some situations a physical item is taken out of the physicalItems array in 
https://github.com/PolymerElements/iron-list/blob/64129116c8b45f1a8db9ba4bab4257b5ffd5c900/iron-list.html#L1757-L1759

Then parent properties are forwarded to the physical items in
https://github.com/PolymerElements/iron-list/blob/64129116c8b45f1a8db9ba4bab4257b5ffd5c900/iron-list.html#L1067-L1086

And evenutally an item that has missed a binding update is returned in to the pool at
https://github.com/PolymerElements/iron-list/blob/64129116c8b45f1a8db9ba4bab4257b5ffd5c900/iron-list.html#L1780
### Expected outcome

All the properties are propagated correclty for out of scope bindings.
### Actual outcome

Some physical items miss binding updates.
### Steps to reproduce

It's not easy to reproduce in an isolated setup. In our application it happens if we select one item
and make it invisible using scrolling.

### Browsers Affected

- [x] Chrome
- [ ] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [ ] Edge
- [ ] IE 11
- [ ] IE 10

### PR DESCRIPTION
**Pull request by sparlampe**
This PR is to demonstrate what fixes issue #340.

### CODE REVIEW

**Comment by sparlampe** [comment]
I signed it!

File: iron-list.html [hunk 1]
```diff
@@ -1065,11 +1065,14 @@
      * notifying parent path change on each row.
      */
     _forwardParentProp: function(prop, value) {
-      if (this._physicalItems) {
-        this._physicalItems.forEach(function(item) {
+
+      var allPhysicalItems = [this._focusedItem,this._offscreenFocusedItem,this._focusBackfillItem]
```
Comment (blasten) [inline iron-list.html]: would you mind changing this to:

```js
_forwardParentProp: function(prop, value) {
  (this._physicalItems || [])
      .concat([this._offscreenFocusedItem, this._focusBackfillItem])
      .forEach(function(item) {
        if (item) {
          item._templateInstance[prop] = value;
        }
      });
}
```

File: iron-list.html [hunk 2]
```diff
@@ -1078,11 +1081,14 @@
      * notifying parent.<path> path change on each row.
      */
     _forwardParentPath: function(path, value) {
-      if (this._physicalItems) {
-        this._physicalItems.forEach(function(item) {
+
+      var allPhysicalItems = [this._focusedItem,this._offscreenFocusedItem,this._focusBackfillItem]
```
Comment (blasten) [inline iron-list.html]: similar comment

**Comment by blasten** [comment]
@sparlampe Sorry for the late feedback. Would you mind rebasing your branch? The "resolve conflicts" feature on GH doesn't seem to work atm.

**Pull request merged**